Sources/ManageMembergroups.php

```diff
@@ -814,6 +849,18 @@ function EditMembergroup()
 							'column' => $board_action == 'allow' ? 'member_groups' : 'deny_member_groups',
 						)
 					);
+
+					$insert = array();
+					foreach ( $changed_boards[$board_action] as $board_id)
+						$insert[] = array((int) $_REQUEST['group'], $board_id, $board_action == 'allow' ? 0 : 1);
+
```

Comment:
Please do the DELETE query here, with a `WHERE id_group = {int:group_id} AND id_board IN ({array_int:boards})` condition.

Reply:
This forearch loop run only for existing boards (where you got an allow or deny),
not for all boards,
exp. when a group lost his allow for board 2, the loop with var  $changed_boards[$board_action] will not included board id 2 since it got deleted

-> i can't place here the delete logic, because the part is only called when you add permission.

@SychO9 could explain it better.